WooCommerce/Classes/ViewRelated/Dashboard/Free Trial/FreeTrialSurvey/FreeTrialSurveyView.swift

```diff
@@ -0,0 +1,123 @@
+import SwiftUI
+
+final class FreeTrialSurveyHostingController: UIHostingController<FreeTrialSurveyView> {
+    init(viewModel: FreeTrialSurveyViewModel) {
+        super.init(rootView: FreeTrialSurveyView(viewModel: viewModel))
+        rootView.dismissAction = dismiss
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureTransparentNavigationBar()
+    }
+
+    func dismiss() {
+        dismiss(animated: true)
+    }
+}
+
+struct FreeTrialSurveyView: View {
+    @ObservedObject private var viewModel: FreeTrialSurveyViewModel
+    var dismissAction: () -> Void = {}
+
+    init(viewModel: FreeTrialSurveyViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        ScrollView {
+            VStack(alignment: .leading, spacing: 40) {
+                Text(Localization.title)
+                    .fontWeight(.bold)
+                    .titleStyle()
+
+                VStack() {
+                    ForEach(viewModel.answers, id: \.self) { answer in
+                        if answer == .otherReasons {
+                            TextField(answer.text, text: $viewModel.otherReasonSpecified)
+                                .font(.body)
+                                .textFieldStyle(RoundedBorderTextFieldStyle(focused: false))
+                        } else {
+                            Button(action: {
+                                viewModel.selectAnswer(answer)
+                            }, label: {
+                                HStack {
+                                    Text(answer.text)
+                                    Spacer()
+                                }
+                            })
+                            .buttonStyle(SelectableSecondaryButtonStyle(isSelected: viewModel.selectedAnswer == answer))
+                        }
+                    }
+                }
+            }
+            .padding(Layout.contentPadding)
+        }
+        .safeAreaInset(edge: .bottom) {
+            VStack(spacing: Layout.ctaPadding) {
+                Divider()
+                    .frame(height: Layout.dividerHeight)
+                    .foregroundColor(Color(.separator))
+
+                Button(Localization.sendFeedback) {
+                    viewModel.submitFeedback()
+                }
+                .buttonStyle(PrimaryButtonStyle())
+                .padding(.horizontal, Layout.ctaPadding)
+                .disabled(!viewModel.feedbackSelected)
+            }
+            .background(Color(.systemBackground))
+        }
+        .toolbar {
+            ToolbarItem(placement: .navigationBarLeading) {
```

Comment:
💡 We can also use `.cancellationAction` and let the system decides where to place the dismiss button. This supports RTL languages more nicely.

Reply:
TIL Done in https://github.com/woocommerce/woocommerce-ios/pull/10319/commits/295b59b170b56172791f1b54b40ff2cf1048e368